litellm/proxy/auth/auth_checks.py

```diff
@@ -263,15 +268,57 @@ async def get_user_object(
             return cached_user_obj
     # else, check db
     try:
-
         response = await prisma_client.db.litellm_usertable.find_unique(
             where={"user_id": user_id}
         )
 
         if response is None:
             if user_id_upsert:
+                # grab the default if exists
+                user_to_add = {"user_id": user_id}
+
+                # check if the user_email_jwt_field is set
+                if user_email:
+                    user_to_add["user_email"] = user_email
+
+                # check if a default team is set, if it is, associated it - create it otherwise
+                if team_name_default:
+                    try:
+                        # if the default team exists, verify it exists, if not upsert it
+                        response = await prisma_client.db.litellm_teamtable.find_unique(
+                            where={"team_alias": team_name_default}
+                        )
+
+                        print("RESPONSE AFTER FINDING TEAM FROM ALIAS")
+                        print(response)
+                        # If we have a response, add the team id
+                        if response:
+                            user_to_add["teams"] = [response.data.team_id]
+
+                    except Exception as e:  # if user not in db
+                        # Create the team id automatically
```

Comment:
the team id upsert logic should not be happening inside 'get_user_object'. this is a generic function, meant to be used for both key + jwt_auth 

and to just get the user object if it exists

Reply:
if team_id upserts need to happen, they should happen within the 'get_team_object' function

Reply:
Moved logic to get_team_object function if the default team name is set and it doesn't exist, it will get created.